lib/kafka/monitor.rb

```diff
@@ -19,8 +21,9 @@ def initialize
     #
     # @param topic [String] The Kafka topic to which the message will be sent
     # @param payload [Hash] The message payload to be sent to the Kafka topic
-    def track_submission_success(topic, payload)
+    def track_submission_success(use_test_topic, payload)
       payload.dig('data', 'ICN') && (payload['data']['ICN'] = '[REDACTED]')
```

Comment:
How to extract the ICN from the payload is actually different depending on the schema of the topic. I would actually suggest leaving the `Kafka::Monitor` methods as they were (accepting the final topic name and payload) and moving the logic to determine the topic and the payload to the calling code, which in this case is the `EventBusSubmissionJob`. You could also add a second method to the topic module that can redact the payload based on the type of topic.

Reply:
Good catch. Will do.